bindings/node.js/Makefile

```diff
@@ -1,40 +1,49 @@
-all: clean build format test bundle
+all: clean build format test
 
 clean:
 	rm -rf build
 	rm -rf dist
+	rm -rf deps
 	rm -f *.node
 	rm -f *.a
 	rm -f *.o
 	rm -rf node_modules
+	rm -rf ref-tests
 
-build: src/kzg.cxx lib/kzg.ts package.json binding.gyp Makefile
-	cd ../../src && make c_kzg_4844.o && cp c_kzg_4844.o ../bindings/node.js
+build: clean src/kzg.cxx lib/kzg.ts package.json binding.gyp Makefile
```

Comment:
I am not sure that I agree with merging the `build` and `bundle` from a JS/TS devEx perspective.

Generally in JS/TS repos there are a few distinct phases for development 
- install
- build/test
- bundle/publish

The install process for node dependencies (`node_modules` folder) is long running and mostly designed to be run once.  The build process is the one that gets iteratively run on file changes (usually there is a watcher monitoring for file changes and rerunning build/test).  Bundle/publish happens infrequently and generally by CI workflows triggered by git tags.

The idea behind structuring them that way is the "build" step is non-existent for JS and the code gets run through the tests with each file save.  For TS, the code is transpiled (in-memory) and then run through the tests similarly.  Because the process happens constantly, to minimize regression, it's ideally as fast as possible.

Native code is a bit different as there is a distinct "build" phase but it's relatively quick for this library. In practice, during active development, I have `chokidar` installed globally, run via a watch script that is not committed, so that I can see when things break with each keystroke.

We are using the publish command on CI to validate cross compatibility.  This is ideal for CI validation before merging but not great during active development.  It is much faster to just rerun node-gyp than to wait for all the node dependencies to be installed on file save.

As a workflow I would love to be able to have `build` running on a loop and then when I'm done with wholesale changes to run the container with `make publish; cd dist; yarn` once to make sure any changes are cross-compatible

As a suggestion, I have copied a Makefile below but please feel free to ignore or use only parts that you like 
```
all: install build format test bundle

# cleans typescript dependencies
clean-install:
	rm -rf node_modules

# installs typescript dependencies
install: clean-install
	yarn install --ignore-scripts

# Cleans native dependency, bindings and typescript artifacts
clean:
	rm -rf build
	rm -rf dist
	rm -rf deps
	rm -f *.node
	rm -f *.a
	rm -f *.o
	rm -rf ref-tests

# Cleans and rebuilds native dependencies, bindings and typescript wrapper
build: clean src/kzg.cxx lib/kzg.ts package.json tsconfig.build.json binding.gyp Makefile
	# Prepare the dependencies directory
	mkdir -p deps/c-kzg
	cp -r ../../blst deps
	cp ../../src/c_kzg_4844.c deps/c-kzg
	cp ../../src/c_kzg_4844.h deps/c-kzg
	# Patch the blst_aux.h to fix a compiler error
	awk '{gsub(/typedef struct \{\} blst_uniq;/, "typedef struct { int dummy; } blst_uniq;")}1' \
		deps/blst/bindings/blst_aux.h > blst_aux_temp.h
	mv blst_aux_temp.h deps/blst/bindings/blst_aux.h
	# Build the bindings
	yarn node-gyp configure
	yarn node-gyp build
	yarn tsc -p tsconfig.build.json

# Bundle the distribution, also helpful for cross compatibility checks
bundle:
	mv deps dist
	cp -r src dist/src
	cp README.md dist/README.md
	cp package.json dist/package.json
	cp binding.gyp dist/binding.gyp

# run unit tests and ref-tests
test:
	yarn jest

# lint js/ts code
format:
	yarn prettier --write .

# publish package to npm. requires an auth token
publish: build
	cd dist
	npm publish

# run ref-tests in linux environment for cross-compatability check
linux-test: build
	# Docker cannot copy from outside this dir
	cp -r ../../tests ref-tests
	docker build -t "linux-test" .
	docker logs --follow `docker run -d linux-test`
	rm -rf ref-tests
```

Reply:
Sounds good, I will split those up again. Your reasoning makes sense.

Reply:
Made these changes, with some alterations.